nx/lib/nx/binary_backend.ex

```diff
@@ -22,8 +22,8 @@ defmodule Nx.BinaryBackend do
   ## Creation
 
   @impl true
-  def scalar(%{type: type, shape: shape} = out, scalar, _backend_options) do
-    data = :binary.copy(number_to_binary(scalar, type), Nx.size(shape))
+  def constant(%{type: type, shape: shape} = out, constant, _backend_options) do
```

Comment:
Why the change from scalar to constant?

Reply:
I think that scalar reflects better the mathematical sense, whereas constant seems to be the opposite of variable (which isn't the case)

Reply:
@polvalente it is a constant because it can have any shape. :)

Reply:
Nice, I missed that :)

Reply:
To be clear, it was previously incorrect, it was always a constant because we use it for optimizations, but I noticed it only now when exposing the API.